src/unix-manager.c

```diff
@@ -702,18 +702,30 @@ TmEcode UnixManagerCaptureModeCommand(json_t *cmd,
     SCReturnInt(TM_ECODE_OK);
 }
 
-TmEcode UnixManagerReloadRules(json_t *cmd, json_t *server_msg, void *data)
+TmEcode UnixManagerReloadRulesWrapper(json_t *cmd, json_t *server_msg, void *data, int wait)
```

Comment:
As this function does not wait for the reload complete, it could be used to (attempt to) start another while the first is still active. DetectEngineReloadStart() will return -1 in this case, which should be handled.

It's something that should be handled in general I think, as we can also have mixes of SIGUSR2 and unix socket leading to this in master.